Wanted to make some initial observations before our Neuro Apero.

vdkreft opened the pull request:
I know I probably made a few mistakes in coding, but I noticed some things that were not explained in the parameters and also some minor typos. I also want to know if you can severely cut down on computation time by removing windows entirely (since you no longer want to pursue that). Let me know what you think of the edits I made!

**Also here are my notes:**

On line 46, it states skip_bids_validation. In comparison to line 161, it seems like you may need a = True statement or = False statement. Am I mistaken?

Line 46: includes task, parameters do not include this variable explanation. (Cs, rest, ns, none)

Line 46: includes mem_mb, parameters do not include this variable explanation.

Line 46: includes sloppy, parameters do not include this variable explanation.

Line 137: comment is a little misleading as it may make one feel that specific if-statement removes both NaNs and infs. Is it better practice to split them into different ifs?

Line 161, 286, 314, 344, 373, 392, 454, 532: Does not include a parameter description for self. Do not know what this does, but have left an open spot for it.

vdkreft committed:
Wanted to make some initial observations before our Neuro Apero.

victoris93 commented on NeuroConn/preprocessing/preprocessing.py:
```diff
@@ -43,7 +43,7 @@ def parse_path_windows_docker(path):
     return path
 
-def parse_fmriprep_command(data_path, fmriprep_path, fs_license_path, work_path, participant_label, skip_bids_validation, nthreads, output_spaces, fs_recon_all, task, niprep_wrapper,mem_mb, sloppy = False, system = platform.system()):
+def parse_fmriprep_command(data_path, fmriprep_path, fs_license_path, work_path, participant_label, skip_bids_validation = True, nthreads, output_spaces, fs_recon_all, task, niprep_wrapper,mem_mb, sloppy = False, system = platform.system()):
```
Ok to skip bids validation by default. I just realized that it should be `nipreps_wrapper` instead of `niprep_wrapper`. A minor thing, but can you go over the code and make this change? just do cmd+F and then make a replacement of all instances of `niprep_wrapper`

victoris93 commented:
> On line 46, it states skip_bids_validation. In comparison to line 161, it seems like you may need a = True statement or = False statement. Am I mistaken?

In fact, it's a question of defaults (Daniel mentioned it on Friday). It's ok I think to skip bids validation by default since we assume the user has already validated their dataset. So yeah, we can leave it as True.

>Line 46: includes task, parameters do not include this variable explanation. (Cs, rest, ns, none)

Yeah, good catch! Note tho that after fmriprep task is not optional.

> Line 46: includes mem_mb, parameters do not include this variable explanation.
> Line 46: includes sloppy, parameters do not include this variable explanation.

Good catch again!

> Line 137: comment is a little misleading as it may make one feel that specific if-statement removes both NaNs and infs. Is it better practice to split them into different ifs?

These comments were a note to self, they are not really of use.

> Line 161, 286, 314, 344, 373, 392, 454, 532: Does not include a parameter description for self. Do not know what this does, but have left an open spot for it.

Remember that it's the class it**self**, it's not an explicit argument.
_____________________

I left some comments, take a look. But once you make the changes, it should be good!

vdkreft committed:
Merge branch 'victoris93:master' into master

vdkreft committed:
Made edits as requested.

Removed self, replaced niprep_wrapper with nipreps_wrapper, and made sure tasks were not
optional.

vdkreft commented:
I have made the specified edits. Let me know if there is anything else you would like for me to do! Bon weekend!

victoris93 commented on NeuroConn/preprocessing/preprocessing.py:
```diff
@@ -43,7 +43,7 @@ def parse_path_windows_docker(path):
     return path
 
-def parse_fmriprep_command(data_path, fmriprep_path, fs_license_path, work_path, participant_label, skip_bids_validation = True, nthreads, output_spaces, fs_recon_all, task, niprep_wrapper,mem_mb, sloppy = False, system = platform.system()):
+def parse_fmriprep_command(data_path, fmriprep_path, fs_license_path, work_path, participant_label, skip_bids_validation = True, nthreads, output_spaces, fs_recon_all, task, nipreps_wrapper,mem_mb, sloppy = False, system = platform.system()):
```
I missed this one the first time, but the default arguments should never be followed by non-default arguments. place skip_bids before sloppy.

victoris93 reviewed (changes requested):
Teaching moment: non-default arguments can not follow default arguments when a function/method is defined.

vdkreft committed:
Merge branch 'victoris93:master' into master
